tests/thandle_ut/thandle_ut.c

```diff
@@ -71,6 +72,50 @@ extern "C" {
 #undef THANDLE_MALLOC_FUNCTION
 #undef THANDLE_FREE_FUNCTION
 
+
+typedef struct A_S_TAG
+{
+    int a;
+    char* s;
+}A_S;
+
+static int copy_A_S(A_S* destination, const A_S* source)
+{
+    int result;
+    destination->a = source->a;
+    destination->s = (char*)malloc(strlen(source->s)+1);
+
+    if (destination->s == NULL)
+    {
+        result = MU_FAILURE;
+    }
+    else
+    {
+        (void)memcpy(destination->s, source->s, strlen(source->s) + 1);
+        result = 0;
+    }
+
+    return result;
+}
+
+static void dispose_A_S(A_S* a_s)
+{
+    free(a_s->s);
+}
+
+#define THANDLE_MALLOC_FUNCTION gballoc_malloc
```

Comment:
An interesting thing about #define and #undef: Recently I tried enabling precompiled headers. And ... #define/#undef - like the ones in umock or here are actually not that nice.
I don't have a better solution now, but maybe we can both think of something better... for the next PR.

Reply:
yup, cannot have precompiled headers that change shape/form :(